src/Services/ZeroRPCFactory.php

```diff
@@ -38,3 +40,44 @@ public function destroyClient($serviceName, $version, $timeout = null)
         unset(self::$_clients[$key]);
     }
 }
+
+/*
+ * When a client object is created, the PHP client will immediately connect
+ * to the endpoint. This wrapper avoids connecting until there is an actual
+ * function call.
+ */
+class _LazyClientWrapper
+{
+    private $_client = null;
+    private $_serviceName;
+    private $_version;
+    private $_context;
+    private $_timeout;
+
+    public function __construct($serviceName, $version, $context, $timeout)
+    {
+        $this->_serviceName = $serviceName;
+        $this->_version = $version;
+        $this->_context = $context;
+        $this->_timeout = $timeout;
+    }
+
+    public function __call(string $name, array $arguments)
```

Comment:
If you want to maintain this package on php 5.4, `string` should be remove.
https://github.com/juwai/laravel-zerorpc/blob/master/composer.json#L18